node/modules/chain.go

```diff
@@ -58,7 +58,7 @@ func ChainBlockService(bs dtypes.ExposedBlockstore, rem dtypes.ChainBitswap) dty
 	return blockservice.New(bs, rem)
 }
 
-func MessagePool(lc fx.Lifecycle, mpp messagepool.Provider, ds dtypes.MetadataDS, nn dtypes.NetworkName, j journal.Journal) (*messagepool.MessagePool, error) {
+func MessagePool(lc fx.Lifecycle, mpp messagepool.Provider, ds dtypes.MetadataDS, nn dtypes.NetworkName, j journal.Journal, protector dtypes.GCReferenceProtector) (*messagepool.MessagePool, error) {
```

Comment:
This isn't really a _dependency_, so I assume this would be a separate "invoke" step (next to the splitstore step). But I don't have strong opinions.

Reply:
Well, I have strong opinions. Just not about this.

Reply:
We could do it with an invoke step, but doing it here avoids all possible races I think.
No strong opinions on this either.

Reply:
basically I want to make sure that `AddProtector` invocations all happen before `Start` in the splitstore.

Reply:
I assume that `Start` gets called from a hook, which runs after all invokes (IIRC).